### ISSUE TITLE
Contrast themes – Buttons appear differently depending on whether they use `a` or `button` element

### ISSUE THREAD

**Issue by Scotchester** [wagtail/wagtail#9078]
### Issue Summary

Our styled buttons are meant to look the same whether they use an `a` element or a `button` element (normal mode screenshot on left). However, when using a forced-colors mode (like Windows High Contrast Mode), they appear differently (Chrome-emulated forced-colors mode screenshot on the right).

| normal | high-contrast |
| ------ | ------------- |
| ![image](https://user-images.githubusercontent.com/1044670/186446656-6e9d0ee4-6639-4a89-aaa9-3589e75ed2e6.png) | ![image](https://user-images.githubusercontent.com/1044670/186446695-65790744-5760-4479-8e82-cb548f2f5101.png) |

### Proposed Solution

We should add some styling in forced-colors mode to give all `button.button`s the same text and border color that `a.button`s get by default.

**Comment by lb-** [comment]
@Scotchester  - we may need to ensure that work on this does not conflict with work on https://github.com/wagtail/wagtail/issues/8790

Happy to chat on Slack but @PaarthAgarwal  is currently looking at doing a bit of a clean up of the button styles first + adding examples to Storybook

Might be good to hold on this one until early next week and we can check in on progress on the clean up side.

**Comment by Scotchester** [comment]
No problem; it's not urgent :)

**Comment by lb-** [comment]
@Scotchester @anujaraj  - the refactoring of the button styles is now in main, should be a lot easier to go in and clean up the high contrast issues now.

Note: be aware of buttons with icons - each has two variations (one for new and one for old icons) and it's a bit of a mess to test.

Do not rely on the styleguide page alone, do smoke tests across a few pages like snippets, modeladmin, page listing etc.

Good luck :D

**Comment by PaarthAgarwal** [comment]
> @Scotchester @anujaraj - the refactoring of the button styles is now in main, should be a lot easier to go in and clean up the high contrast issues now.
> 
> #9104
> 
> Note: be aware of buttons with icons - each has two variations (one for new and one for old icons) and it's a bit of a mess to test.
> 
> Do not rely on the styleguide page alone

Or on Pattern Library too ;)

**Comment by albinazs** [comment]
Hello @thibaudcolas! I would like to work on this issue

**Comment by thibaudcolas** [comment]
@albinazs you can go for this without checking with anyone else – but note since you’ve also claimed #8659, it’d be nice for you to either finish that one, or comment there that you no longer intend to work on it. We can’t let people claim all the issues 😛

**Comment by albinazs** [comment]
Dear mentors, @thibaudcolas @lb- 
I managed to solve the issue with the following piece of code in `components/_button.scss`:

```@media (forced-colors: active) {
  .button {
    color: LinkText;
    border-color: LinkText;
    &.button-secondary {
      color: LinkText;
    }
    &.no,
    &.serious {
      color: LinkText;
      border-color: LinkText;
    }
    &.bicolor {
      &.button-secondary {
        border-color: LinkText;
      }
    }
    &.text-replace {
      &.button--icon {
        border-color: LinkText;
      }
    }
  }
}
```

Here I basically overwrite in a separate place (for more readability) all button classes with either `color` or `border-color` specified. 
The result in Storybook is like this:

| Before  | After |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/51043550/195554818-b67e3775-79e0-460b-8c53-ed7a1a2a4451.png)  | ![image](https://user-images.githubusercontent.com/51043550/195554938-5f7b83b7-eb05-44ab-a83a-89add8329a94.png) |

As advised on slack, I only used Chrome dev tools simulator to test it.
Can you please guide me on any mistakes I made and on my next steps - how shall I test my solution further (let say how to set up new Wagtail site with the changes I made in my repo branch) - or any other ideas.

In the process I also noticed another issue:  in WHCM dark Theme buttons with icons only look empty:

<img width="204" alt="image" src="https://user-images.githubusercontent.com/51043550/195556969-02f4a539-6497-4da7-80d4-2a260c187447.png">

<img width="75" alt="image" src="https://user-images.githubusercontent.com/51043550/195557018-986f71da-3949-4a66-8dae-bc2b96c3884d.png">

**Comment by lb-** [comment]
@albinazs  a quick tip - you can test (for a few mins) WHCM on assistiv labs https://assistivlabs.com/assistive-tech/display/high-contrast-mode but if you need more robust testing you will need to run a VM locally (Happy to help but ask on Slack in the Outreachy channel).

If you want to quickly check something on the latest version of Wagtail develop `main` you can also do that with this URL https://wagtaildemo-nightly.herokuapp.com/admin/

**Comment by albinazs** [comment]
Got it, @lb-.  I started with LinkText because it was a proposed solution in an issue, but I agree that ButtonText makes more sense.

I made tests via Assistive Labs on all 4 WHCMs available in 21H2 version, both Edge & Chrome. The static buttons look good (see screenshots below).
But the real fun starts with hovering:) Elements behave very differently, so now I will be fixing this one as well.

Testing via Assistive Labs also confirmed the issue with icons-only buttons. I will check later on the actual Windows device, and open a new issue accordingly.

High Contrast Black

![image](https://user-images.githubusercontent.com/51043550/195619176-80d95bef-5992-4d74-96a6-db5cd09723db.png)

High Contrast White

![image](https://user-images.githubusercontent.com/51043550/195619333-9165819a-66bc-4f8b-95b6-7087c2f016dd.png)

High Contrast #1

![image](https://user-images.githubusercontent.com/51043550/195619437-3de1ca30-7072-42f4-a44b-b55a137a73bb.png)

High Contrast #2

![image](https://user-images.githubusercontent.com/51043550/195619533-9fcb2bce-8a6c-4714-a870-5b3055762e0d.png)

**Comment by lb-** [comment]
@albinazs let's go with ButtonTexr - great solution. Let's get a PR up and we can move ahead with this.
